### PR TITLE
simplified upgrade command, split upgrade API

### PR DESCRIPTION
**Pull request by dmitsh**
- simplify upgrade command to specify desired kubernetes version in the command line
- split upgrade API (20170930) into cluster dependent (20170701, 20170831) and cluster independent (20170930)

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/api/agentPoolOnlyApi/v20170831/types.go [hunk 1]
```diff
@@ -94,6 +94,26 @@ const (
 	Migrating ProvisioningState = "Migrating"
 )
 
+// PoolUpgradeProfile contains pool properties:
+//  - kubernetes version
+//  - pool name (for agent pool)
+//  - OS type of the VMs in the pool
+//  - list of applicable upgrades
+type PoolUpgradeProfile struct {
```
Comment (weinong) [inline pkg/api/agentPoolOnlyApi/v20170831/types.go]: Can we name it AgentPoolUpgradeProfile instead?

Comment (weinong) [reply pkg/api/agentPoolOnlyApi/v20170831/types.go]: Nvm. Looks like it's used by ControlPlane as well as by AgentPool

Comment (dmitsh) [reply pkg/api/agentPoolOnlyApi/v20170831/types.go]: It also works for master/controlPlane pool

File: pkg/api/v20170701/types.go [hunk 2]
```diff
@@ -184,6 +184,26 @@ type AgentPoolProfile struct {
...
+	Name     string                 `json:"name,omitempty"`
+	OSType   string                 `json:"osType,omitempty"`
+	Upgrades []*OrchestratorProfile `json:"upgrades,omitempty"`
+}
+
+// UpgradeProfile contains cluster properties:
+//  - orchestrator type and version for the cluster
+//  - list of pool profiles, constituting the cluster
+type UpgradeProfile struct {
+	ControlPlaneProfile *PoolUpgradeProfile   `json:"controlPlaneProfile"`
```
Comment (weinong) [inline pkg/api/v20170701/types.go]: isn't this HCP only?

File: pkg/api/agentPoolOnlyApi/v20170831/types.go [hunk 3]
```diff
@@ -94,6 +94,26 @@ const (
...
+	KubernetesVersion string    `json:"kubernetesVersion"`
+	Name              string    `json:"name,omitempty"`
+	OSType            string    `json:"osType,omitempty"`
+	Upgrades          []*string `json:"upgrades,omitempty"`
+}
+
+// UpgradeProfile contains cluster properties:
```
Comment (weinong) [inline pkg/api/agentPoolOnlyApi/v20170831/types.go]: i'm not sure if this comment reflects this type correctly.

File: pkg/api/v20170701/types.go [hunk 4]
```diff
@@ -184,6 +184,26 @@ type AgentPoolProfile struct {
 	subnet string
 }
 
+// PoolUpgradeProfile contains pool properties:
```
Comment (weinong) [inline pkg/api/v20170701/types.go]: does it mean each pool may have different orchestrators?

Comment (dmitsh) [reply pkg/api/v20170701/types.go]: Each pool may potentially have different orchestrator versions. This is not implemented yet.

File: pkg/api/v20170701/types.go (hunk 2 above)
Comment (dmitsh) [reply pkg/api/v20170701/types.go]: Oh, I see. Will change.

File: pkg/api/agentPoolOnlyApi/v20170831/types.go (hunk 3 above)
Comment (dmitsh) [reply pkg/api/agentPoolOnlyApi/v20170831/types.go]: addressed

**Pull request merged**